Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm

```diff
@@ -1747,8 +1747,8 @@ llintOpWithMetadata(op_get_by_val, OpGetByVal, macro (size, get, dispatch, metad
     jmp .opGetByValNotEmpty
 
 .opGetByValNotDouble:
-    subi ArrayStorageShape, t2
-    bia t2, SlowPutArrayStorageShape - ArrayStorageShape, .opGetByValNotIndexedStorage
+    bieq t2, AlwaysSlowPutContiguousShape, .opGetByValIsContiguous
+    bilt t2, ArrayStorageShape, .opGetByValNotIndexedStorage
```

Comment:
Why does the old code not work?

Reply:
It does work correctly, but takes a slow path for new indexing shape. The proposed change leverages the fact that `SlowPutArrayStorageShape` is the last indexing shape (arithmetically) to maintain the same # of opcodes while ensuring `AlwaysSlowPutContiguousShape` is fast even in LLInt.